src/ws/ws_manager.rs

```diff
@@ -166,32 +170,74 @@ impl WsManager {
-            serde_json::from_str::<Message>(&data).map_err(|e| Error::JsonParse(e.to_string()))?;
-        let identifier = WsManager::get_identifier(&message)?;
-        if identifier.is_empty() {
-            return Ok(());
+        match data {
+            Some(data) => match data {
+                Ok(data) => match data.into_text() {
+                    Ok(data) => {
+                        if !data.starts_with('{') {
+                            return Ok(());
+                        }
+                        let message = serde_json::from_str::<Message>(&data)
+                            .map_err(|e| Error::JsonParse(e.to_string()))?;
+                        let identifier = WsManager::get_identifier(&message)?;
+                        if identifier.is_empty() {
+                            return Ok(());
+                        }
+
+                        let mut subscriptions = subscriptions.lock().await;
+                        let mut res = Ok(());
+                        if let Some(subscription_datas) = subscriptions.get_mut(&identifier) {
+                            for subscription_data in subscription_datas {
+                                if let Err(e) = subscription_data
+                                    .sending_channel
+                                    .send(message.clone())
+                                    .map_err(|e| Error::WsSend(e.to_string()))
+                                {
+                                    res = Err(e);
+                                }
+                            }
+                        }
+                        res
+                    }
+                    Err(err) => {
+                        let error = Error::ReaderTextConversion(err.to_string());
+                        Ok(WsManager::send_to_all_subscriptions(
+                            subscriptions,
+                            Message::HyperliquidError(error.to_string()),
+                        )
+                        .await?)
+                    }
+                },
+                Err(err) => {
+                    let error = Error::GenericReader(err.to_string());
+                    Ok(WsManager::send_to_all_subscriptions(
+                        subscriptions,
+                        Message::HyperliquidError(error.to_string()),
+                    )
+                    .await?)
+                }
+            },
+            None => Ok(WsManager::send_to_all_subscriptions(subscriptions, Message::NoData).await?),
         }
+    }
 
+    async fn send_to_all_subscriptions(
+        subscriptions: &Arc<Mutex<HashMap<String, Vec<SubscriptionData>>>>,
+        message: Message,
+    ) -> Result<()> {
         let mut subscriptions = subscriptions.lock().await;
         let mut res = Ok(());
-        if let Some(subscription_datas) = subscriptions.get_mut(&identifier) {
-            for subscription_data in subscription_datas {
+        for (_, subs_data) in subscriptions.iter_mut() {
```

Comment:
consistency in naming - `subscriptions_data`, also just use `values_mut` if you only care about the values